src/operator/nn/batch_norm.cc

```diff
@@ -420,10 +420,14 @@ static bool BatchNormType(const nnvm::NodeAttrs& attrs,
 
 #if MXNET_USE_MKLDNN == 1
 static inline bool SupportMKLDNNBN(const NDArray &input, const BatchNormParam &param) {
-  mxnet::TShape shape = input.shape();
-  return SupportMKLDNN(input) && shape.ndim() == 4
```

Comment:
we are removing the check for `ndim == 4` here, and another lighter check for `ndim == 1 || ndim ==2 || ndim ==4` present in `SupportMKLDNN`.
Does that mean `ndim` can be anything >0 ? What are the allowed values for `ndim`?

Reply:
Yes, `ndim` coulld be anything > 0.

If the shape of input A is `shape`, the input will be shaped into `(prod(shape[0:axis]), shape[axis], 1, prod(shape[axis+1:len(shape)]) )`.

ndim|shape|axis=0|axis=1|axis=2|axis=3|axis=4
---|---|--|--|--|--|---
1|(N,)|(1, N, 1, 1)|x|x|x|x
2|(N,C)|(1, N, 1, C)|(N, C, 1, 1)|x|x|x
3|(N,C,H)|(1, N, 1, C*H)|(N, C, 1, H)|(N*C, H, 1, 1)|x|x
4|(N,C,H,W)|(1, N, 1, C*H*W)|(N, C, 1, H*W)|(N*C, H, 1, W)|(N*C*H, W, 1, 1)|x
5|(N,D,C,H,W)|(1, N, 1, D*C*H*W)|(N, D, 1, C*H*W)|(N*D,C, 1, H*W)|(N*D*C, H, 1, W)|(N*D*C*H, W, 1, 1)

Reply:
Thanks for the explanation! Does the table continue further for ndims > 5? Or should we place a check for that?

Reply:
Yes, it supports ndim > 5 too.